contracts/LBP/LBPWrapper.sol

```diff
@@ -0,0 +1,119 @@
+/*
+██████╗░██████╗░██╗███╗░░░███╗███████╗██████╗░░█████╗░░█████╗░
+██╔══██╗██╔══██╗██║████╗░████║██╔════╝██╔══██╗██╔══██╗██╔══██╗
+██████╔╝██████╔╝██║██╔████╔██║█████╗░░██║░░██║███████║██║░░██║
+██╔═══╝░██╔══██╗██║██║╚██╔╝██║██╔══╝░░██║░░██║██╔══██║██║░░██║
+██║░░░░░██║░░██║██║██║░╚═╝░██║███████╗██████╔╝██║░░██║╚█████╔╝
+╚═╝░░░░░╚═╝░░╚═╝╚═╝╚═╝░░░░░╚═╝╚══════╝╚═════╝░╚═╝░░╚═╝░╚════╝░
+*/
+
+// SPDX-License-Identifier: GPL-3.0-or-later
+
+// solium-disable linebreak-style
+pragma solidity ^0.8.4;
+
+
+import "@openzeppelin/contracts/token/ERC20/IERC20.sol";
+import "../utils/interface/ILBPFactory.sol";
+import "../utils/interface/IVault.sol";
+import "../utils/interface/ILBP.sol";
+import "hardhat/console.sol";
+
+
+contract LBPWrapper {
+
+    address public owner;
+    address public LBPFactory;
+    address public lbp;
+    uint256 public swapFeePercentage;
+    bool isInitialized;
+
+    modifier onlyOwner{
+        require(msg.sender == owner, "LBPWrapper: only owner function");
+        _;
+    }
+
+    function transferOwnership(address _newOwner) public onlyOwner{
+        require(_newOwner != address(0), "LBPWrapper: new owner cannot be zero");
+        owner = _newOwner;
+    }
+
+    function initialize (
+            address _LBPFactory,
+            uint256 _swapFeePercentage
+    ) public
+    {
+        owner = msg.sender;
+        LBPFactory = _LBPFactory;
+        swapFeePercentage = _swapFeePercentage;
+        isInitialized = true;
+    }
+
+    function deployLbpFromFactory(
+            string memory _name,
+            string memory _symbol,
+            IERC20[] memory _tokens,
+            uint256[] memory _amounts,
+            uint256[] memory _weights,
+            bool _swapEnabledOnStart,
+            uint256 _startTime,
+            uint256 _endTime,
+            uint256[] memory _endWeights,
+            address _admin,
+            bytes memory _userData
+    ) public onlyOwner returns(address)
+    {
+        {
+            address vault = address(ILBPFactory(LBPFactory).getVault());
+            for ( uint i; i < _tokens.length; i++ ) {
+                IERC20(_tokens[i]).transferFrom(msg.sender, address(this), _amounts[i]);
```

Comment:
Do we need to transfer the tokens to the contract? In the example in the [docs](https://docs.balancer.fi/developers/guides/pool-creation)(bottom of page), we only need to approve.
If I misunderstand and we do need to transfer, would in this case be `msg.sender` be the wrapper contract instead of the admin who has the funds?

Reply:
Yes, we are transferring it to wrapper and wrapper approves it for vaul.

Reply:
But now, while typing thi I just realised we don't need this much.

Reply:
Directly approve vault from admin.